lib/buffer.js

```diff
@@ -1108,6 +1108,24 @@ Buffer.prototype.write = function write(string, offset, length, encoding) {
     }
   }
 
+  const len = string.length;
+  if (len <= 16 && len <= length) {
+    let n = 0;
+    while (true) {
+      const code = StringPrototypeCharCodeAt(string, n);
+      if (code >= 128) {
+        break;
+      }
+
+      this[offset + n] = code;
```

Comment:
It seems that the `encoding` argument is ignores so something like `buf.write('fg', 0, 2, 'hex')` will write 2 bytes instead of 0.

Reply:
PTAL